src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs

```diff
@@ -1808,6 +1808,28 @@ checkCaseBranch CaseBranch {..} = withLocalScope $ do
         (p ^. patternArgIsImplicit == Implicit)
         (throw (ErrCaseBranchImplicitPattern (CaseBranchImplicitPattern p)))
 
+checkNewCaseBranch ::
+  forall r.
+  (Members '[Reader ScopeParameters, Error ScoperError, State Scope, State ScoperState, InfoTableBuilder, NameIdGen] r) =>
+  NewCaseBranch 'Parsed ->
+  Sem r (NewCaseBranch 'Scoped)
+checkNewCaseBranch NewCaseBranch {..} = withLocalScope $ do
+  pattern' <- checkParsePatternAtoms _newCaseBranchPattern
+  checkNotImplicit pattern'
+  expression' <- (checkParseExpressionAtoms _newCaseBranchExpression)
+  return $
+    NewCaseBranch
+      { _newCaseBranchPattern = pattern',
+        _newCaseBranchExpression = expression',
+        ..
+      }
+  where
+    checkNotImplicit :: PatternArg -> Sem r ()
+    checkNotImplicit p =
+      when
+        (p ^. patternArgIsImplicit == Implicit)
+        (throw (ErrCaseBranchImplicitPattern (CaseBranchImplicitPattern p)))
```

Comment:
I see that we have no negative test for this error. However, I'm not sure if this error should exist, since this will be caught by the typechecker anyway. What do you think?

Reply:
For me it depends on how good the type checker error is. 

Do you know how do you trigger this error?

Reply:
The error message for
```
   case b of {
      | {{true}} := false
```
is 
```
Expected an explicit pattern but found an implicit instance pattern: {{true}}
```

Reply:
I agree that this error should not exist as it duplicates the error thrown by the arity checker

Reply:
Raised the PR in https://github.com/anoma/juvix/pull/2357